Update `long-format-table-utils/annotator` for v8 data release

logstar opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

Update `long-format-table-utils/annotator` for v8 data release.

#### What was your approach?

The preliminary v8 `ensg-hugo-rmtl-mapping.tsv` in <https://github.com/d3b-center/D3b-codes/pull/48> is used to prepare this PR, because it is probably going to be the v8 release `ensg-hugo-rmtl-mapping.tsv`.

This PR will be updated after v8 release PR <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/89> is merged. This PR is prepared in advance of the release, in order to quickly update `long-format-table-utils/annotator` after v8 is released, so analysis modules can use annotator to generate v8 JSONL files.

The `ensg-hugo-rmtl-mapping.tsv` used in this PR has ENSG IDs mapped to different gene symbols, whereas v7 `ensg-hugo-rmtl-mapping.tsv` has every ENSG ID mapped to only one gene symbol.

```text
> library(tidyverse)
> v8_ehdf <- read_tsv('ensg-hugo-rmtl-mapping.tsv')
> v8_ehdf %>%
+   filter(!is.na(ensg_id), !is.na(gene_symbol)) %>%
+   group_by(ensg_id) %>%
+   summarise(n_uniq_mapped_symbols = length(unique(gene_symbol)),
+             uniq_symbol_list = paste(unique(gene_symbol), collapse = ',')) %>%
+   filter(n_uniq_mapped_symbols > 1)
# A tibble: 30,626 x 3
   ensg_id         n_uniq_mapped_symbols uniq_symbol_list  
   <chr>                           <int> <chr>             
 1 ENSG00000005059                     2 CCDC109B,MCUB     
 2 ENSG00000005189                     2 AC004381.6,REXO5  
 3 ENSG00000005379                     2 BZRAP1,TSPOAP1    
 4 ENSG00000006015                     2 C19orf60,REX1BD   
 5 ENSG00000008197                     2 RP3-336H9.2,TFAP2D
 6 ENSG00000010165                     2 METTL13,EEF1AKNMT 
 7 ENSG00000014257                     2 ACPP,ACP3         
 8 ENSG00000018607                     2 AC097532.4,ZNF806 
 9 ENSG00000018610                     2 CXorf56,STEEP1    
10 ENSG00000020219                     2 CCT8L1,CCT8L1P    
# … with 30,616 more rows
> 
> v7_ehdf <- read_tsv('../../data/ensg-hugo-rmtl-mapping.tsv')
> v7_ehdf %>%
+   filter(!is.na(ensg_id), !is.na(gene_symbol)) %>%
+   group_by(ensg_id) %>%
+   summarise(n_uniq_mapped_symbols = length(unique(gene_symbol)),
+             uniq_symbol_list = paste(unique(gene_symbol), collapse = ',')) %>%
+   filter(n_uniq_mapped_symbols > 1)
# A tibble: 0 x 3
# … with 3 variables: ensg_id <chr>, n_uniq_mapped_symbols <int>, uniq_symbol_list <chr>
```

Therefore, the assumption in previous versions of `annotator` that one ENSG ID is mapped to only one gene symbol no longer holds. In order to handle ENSG IDs that are mapped to different symbols, following changes are made in commit <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/commit/647bc5581cd74cdddaf4659ba02230484277a34f>.

- In `annotator/download-annotation-data.R`:
  - Remove the assertion that all ENSG IDs are unique.
  - Use `unique(ensg_hugo_rmtl_df$ensg_id)` for querying mygene.info for `Gene_full_name` and `Protein_RefSeq_ID`.
- In `annotator/annotator-api.R`, add `dplyr::distinct()` at the end of preparing RMTL annotation table `ann_tbl_l$fda_rmtl$tibble`.

These changes will not affect analysis modules that use annotator, because ENSG IDs are still unique in `annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv` and `annotator-api.R variable ann_tbl_l$fda_rmtl$tibble`, which are the only annotation tables that use ENSG ID column for joining data tables.

These changes are tested as following.

- `run-update-long-format-table-utils.sh` runs well. `annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv` is updated with new ENSG IDs in the preliminary v8 `ensg-hugo-rmtl-mapping.tsv`.
- All unit test passed in the OpenPedCan docker image/container.
- v7 `snv-frequencies` results are reproduced identically.

#### What GitHub issue does your pull request address?

<https://github.com/PediatricOpenTargets/ticket-tracker/issues/172>

### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?

Could the R code changes in this PR break any user interfaces?

#### Is there anything that you want to discuss further?

No.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes.

### Results

#### What types of results are included (e.g., table, figure)?

Table.

#### What is your summary of the results?

```text
$ git diff --stat dev -- annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv
 analyses/long-format-table-utils/annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv | 282 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 282 insertions(+)
```

#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [NA] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

ewafula reviewed (approved):
Thank you @logstar for updating the `long-format-table-utils` module!  I regenerating the annotation table, `ensg-gene-full-name-refseq-protein.tsv` using the the updated `ensg-hugo-rmtl-mapping.tsv` table from the  [D3b-codes](https://github.com/d3b-center/D3b-codes/blob/cb9ae7fe84e709958cc192784525323b9924c6be/gene_match/results/ensg-hugo-rmtl-mapping.tsv) repository and obtained identical results to those in PR. Also tested the module with the downstream  module, `cnv-frequencies` to re-annotate the copy number variant frequencies and obtained similar annotations to the previous results. The updates did not break the downstream analyses.

logstar commented:
> Thank you @logstar for updating the `long-format-table-utils` module! I regenerating the annotation table, `ensg-gene-full-name-refseq-protein.tsv` using the the updated `ensg-hugo-rmtl-mapping.tsv` table from the [D3b-codes](https://github.com/d3b-center/D3b-codes/blob/cb9ae7fe84e709958cc192784525323b9924c6be/gene_match/results/ensg-hugo-rmtl-mapping.tsv) repository and obtained identical results to those in PR. Also tested the module with the downstream module, `cnv-frequencies` to re-annotate the copy number variant frequencies and obtained similar annotations to the previous results. The updates did not break the downstream analyses.

Thank you for reviewing and test-running this PR @ewafula !

I have also rerun `analyses/cnv-frequencies/run-cnv-frequencies-analysis.sh` with the updated annotator in this PR and the preliminary v8 `ensg-hugo-rmtl-mapping.tsv`. The contents of result files, `gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz` and `gene-level-cnv-consensus-annotated-mut-freq.tsv.gz`, are identical to previous ones. The differences are caused by gzip, because gzip adds time-stamp to the zipped file by default. If your `cnv-frequencies` rerun result contents are different from previous ones, could you comment with their `diff`s? Their `diff`s would be informative to check whether their differences are expected or not.

```text
# On the latest git commit of this PR
$ git log -n 1
commit dd8194fd4b6d9b9044aa09b8cf6a2e0774786230 (HEAD -> v8-annotator, origin/v8-annotator)
Author: logstar <y.will.zhang@gmail.com>
Date:   Fri Aug 20 14:48:02 2021 -0400

    Update README.md
    
    Change last update date.
    
    Add a note on duplicated gene ENSG ID to symbol mappings.

# Use the preliminary v8 ensg-hugo-rmtl-mapping.tsv
$ wc -l data/ensg-hugo-rmtl-mapping.tsv
94842 data/ensg-hugo-rmtl-mapping.tsv

# Use the updated ensg-gene-full-name-refseq-protein.tsv of this PR
$ wc -l analyses/long-format-table-utils/annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv 
41475 analyses/long-format-table-utils/annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv

# Rerun run-cnv-frequencies-analysis.sh
$ rm analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.*gz && bash analyses/cnv-frequencies/run-cnv-frequencies-analysis.sh
Read analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_autosomes_freq.tsv...
Annotate analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_autosomes_freq.tsv...
Output analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_autosomes_annot_freq.tsv...
Done.
Read analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_x_and_y_freq.tsv...
Annotate analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_x_and_y_freq.tsv...
Output analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_x_and_y_annot_freq.tsv...
Done.

# gz files are different from previous ones
$ git status
On branch v8-annotator
Your branch is up to date with 'origin/v8-annotator'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz
        modified:   analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.tsv.gz

no changes added to commit (use "git add" and/or "git commit -a")

# gz file contents are identical to previous ones
$ diff <(git show dev:./analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.tsv.gz | gunzip -c) <(gunzip -c analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.tsv.gz) | wc -l
0

$ diff <(git show dev:./analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz | gunzip -c) <(gunzip -c analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz) | wc -l
0
```

ewafula commented:
> > Thank you @logstar for updating the `long-format-table-utils` module! I regenerating the annotation table, `ensg-gene-full-name-refseq-protein.tsv` using the the updated `ensg-hugo-rmtl-mapping.tsv` table from the [D3b-codes](https://github.com/d3b-center/D3b-codes/blob/cb9ae7fe84e709958cc192784525323b9924c6be/gene_match/results/ensg-hugo-rmtl-mapping.tsv) repository and obtained identical results to those in PR. Also tested the module with the downstream module, `cnv-frequencies` to re-annotate the copy number variant frequencies and obtained similar annotations to the previous results. The updates did not break the downstream analyses.
> 
> Thank you for reviewing and test-running this PR @ewafula !
> 
> I have also rerun `analyses/cnv-frequencies/run-cnv-frequencies-analysis.sh` with the updated annotator in this PR, the preliminary v8 `ensg-hugo-rmtl-mapping.tsv`, and v7 CNV data files. The contents of result files, `gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz` and `gene-level-cnv-consensus-annotated-mut-freq.tsv.gz`, are identical to previous ones. The differences are caused by gzip, because gzip adds time-stamp to the zipped file by default. If your `cnv-frequencies` rerun result contents are different from previous ones, could you comment with their `diff`s? Their `diff`s would be informative to check whether their differences are expected or not.
> 
> ```
> # On the latest git commit of this PR
> $ git log -n 1
> commit dd8194fd4b6d9b9044aa09b8cf6a2e0774786230 (HEAD -> v8-annotator, origin/v8-annotator)
> Author: logstar <y.will.zhang@gmail.com>
> Date:   Fri Aug 20 14:48:02 2021 -0400
> 
>     Update README.md
>     
>     Change last update date.
>     
>     Add a note on duplicated gene ENSG ID to symbol mappings.
> 
> # Use the preliminary v8 ensg-hugo-rmtl-mapping.tsv
> $ wc -l data/ensg-hugo-rmtl-mapping.tsv
> 94842 data/ensg-hugo-rmtl-mapping.tsv
> 
> # Use the updated ensg-gene-full-name-refseq-protein.tsv of this PR
> $ wc -l analyses/long-format-table-utils/annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv 
> 41475 analyses/long-format-table-utils/annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv
> 
> # Use v7 CNV data files
> $ ls -lh data/consensus_wgs_plus_cnvkit_wxs_*gz
> lrwxrwxrwx 1 ubuntu ubuntu 49 Aug 20 18:07 data/consensus_wgs_plus_cnvkit_wxs_autosomes.tsv.gz -> v7/consensus_wgs_plus_cnvkit_wxs_autosomes.tsv.gz
> lrwxrwxrwx 1 ubuntu ubuntu 47 Aug 20 18:07 data/consensus_wgs_plus_cnvkit_wxs_x_and_y.tsv.gz -> v7/consensus_wgs_plus_cnvkit_wxs_x_and_y.tsv.gz
> 
> $ md5sum data/consensus_wgs_plus_cnvkit_wxs_*gz
> 28fd94db766a75dba1476a0540b04a74  data/consensus_wgs_plus_cnvkit_wxs_autosomes.tsv.gz
> c14747ff627f04926d5aa3803d4bbca2  data/consensus_wgs_plus_cnvkit_wxs_x_and_y.tsv.gz
> 
> # Rerun run-cnv-frequencies-analysis.sh
> $ rm analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.*gz && bash analyses/cnv-frequencies/run-cnv-frequencies-analysis.sh
> Read analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_autosomes_freq.tsv...
> Annotate analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_autosomes_freq.tsv...
> Output analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_autosomes_annot_freq.tsv...
> Done.
> Read analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_x_and_y_freq.tsv...
> Annotate analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_x_and_y_freq.tsv...
> Output analyses/cnv-frequencies/results/consensus_wgs_plus_cnvkit_wxs_x_and_y_annot_freq.tsv...
> Done.
> 
> # gz files are different from previous ones
> $ git status
> On branch v8-annotator
> Your branch is up to date with 'origin/v8-annotator'.
> 
> Changes not staged for commit:
>   (use "git add <file>..." to update what will be committed)
>   (use "git restore <file>..." to discard changes in working directory)
>         modified:   analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz
>         modified:   analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.tsv.gz
> 
> no changes added to commit (use "git add" and/or "git commit -a")
> 
> # gz file contents are identical to previous ones
> $ diff <(git show dev:./analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.tsv.gz | gunzip -c) <(gunzip -c analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.tsv.gz) | wc -l
> 0
> 
> $ diff <(git show dev:./analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz | gunzip -c) <(gunzip -c analyses/cnv-frequencies/results/gene-level-cnv-consensus-annotated-mut-freq.jsonl.gz) | wc -l
> 0
> ```

@logstar, I used v8 histologies, and my up-to-date synced v7 branch from dev. Any changes in the either `independent- samples module` and/or the histologies file are expected to affect the cnv frequencies. Since the cnv frequencies are generated by functions independent of the function call `long-format-table-utils` to annotate ENS IDs, my intention was not to focus on diffs in the frequency column.  
```
cut -f 13-18 results/gene-level-cnv-consensus-annotated-mut-freq.tsv > test.new.tsv 
cut -f 13-18 results.old/gene-level-cnv-consensus-annotated-mut-freq.tsv > test.old.tsv 

cnv-frequencies]$ wc -l test.new.tsv test.old.tsv 
 559247 test4.txt
 559247 test3.txt
 1118494 total

cnv-frequencies]$ diff test.new.tsv  test.old.tsv  
cnv-frequencies]$ => silent no diffs
````
I'd see the relevance of delving into the difference in cnv frequencies  between v8 and v7 in this PR
```
cnv-frequencies]$ diff results/gene-level-cnv-consensus-annotated-mut-freq.tsv results.old/gene-level-cnv-consensus-annotated-mut-freq.tsv | head 
79061c79061
< SCYL3	ENSG00000000457	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	5/13	38.46%	SCY1 like pseudokinase 3		N	EFO_1000026	MONDO_0006033
---
> SCYL3	ENSG00000000457	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	4/13	30.77%	SCY1 like pseudokinase 3		N	EFO_1000026	MONDO_0006033
79064c79064
< C1orf112	ENSG00000000460	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	5/13	38.46%	chromosome 1 open reading frame 112	EFO_1000026	MONDO_0006033
---
> C1orf112	ENSG00000000460	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	4/13	30.77%	chromosome 1 open reading frame 112	EFO_1000026	MONDO_0006033
79070c79070
````

logstar commented:
> @logstar, I used v8 histologies, and my up-to-date synced v7 branch from dev. Any changes in the either `independent- samples module` and/or the histologies file are expected to affect the cnv frequencies. Since the cnv frequencies are generated by functions independent of the function call `long-format-table-utils` to annotate ENS IDs, my intention was not to focus on diffs in the frequency column.
> 
> ```
> cut -f 13-18 results/gene-level-cnv-consensus-annotated-mut-freq.tsv > test.new.tsv 
> cut -f 13-18 results.old/gene-level-cnv-consensus-annotated-mut-freq.tsv > test.old.tsv 
> 
> cnv-frequencies]$ wc -l test.new.tsv test.old.tsv 
>  559247 test4.txt
>  559247 test3.txt
>  1118494 total
> 
> cnv-frequencies]$ diff test.new.tsv  test.old.tsv  
> cnv-frequencies]$ => silent no diffs
> ```
> 
> I'd see the relevance of delving into the difference in cnv frequencies between v8 and v7 in this PR
> 
> ```
> cnv-frequencies]$ diff results/gene-level-cnv-consensus-annotated-mut-freq.tsv results.old/gene-level-cnv-consensus-annotated-mut-freq.tsv | head 
> 79061c79061
> < SCYL3	ENSG00000000457	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	5/13	38.46%	SCY1 like pseudokinase 3		N	EFO_1000026	MONDO_0006033
> ---
> > SCYL3	ENSG00000000457	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	4/13	30.77%	SCY1 like pseudokinase 3		N	EFO_1000026	MONDO_0006033
> 79064c79064
> < C1orf112	ENSG00000000460	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	5/13	38.46%	chromosome 1 open reading frame 112	EFO_1000026	MONDO_0006033
> ---
> > C1orf112	ENSG00000000460	gain		PBTA	Diffuse midline glioma	15/54	27.78%	8/45	17.78%	4/13	30.77%	chromosome 1 open reading frame 112	EFO_1000026	MONDO_0006033
> 79070c79070
> ```

Thank you for checking, @ewafula . It is expected that `cnv-frequencies` results generated using v8 data will be different from v7 results.

I agree that checking the differences between v7 and v8 `cnv-frequencies` results is out of the scope of this PR.

logstar commented:
I have merged the latest `dev` branch to this PR, in order to test the `annotator` updates using v8 release data.

- `run-update-long-format-table-utils.sh` reproduced `analyses/long-format-table-utils/annotator/annotation-data/ensg-gene-full-name-refseq-protein.tsv` identically.
- All unit tests passed. `analyses/long-format-table-utils/annotator/run-tests.sh`.
- `analyses/snv-frequencies/run-snv-frequencies.sh` runs well. The v8 results are different from v7 results, which is expected.

logstar commented:
Hi @ewafula @komalsrathi @dmiller15 - I will merge this PR soon, because it is required to generate v8 JSONL files. Let me know if you have any review in progress, question, suggestion, or comment.

cc @jharenza

ewafula commented:
@logstar, I don’t have any additional reviews.

On Tue, Aug 24, 2021 at 12:19 PM Yuanchao Zhang ***@***.***>
wrote:

> Hi @ewafula <https://github.com/ewafula> @komalsrathi
> <https://github.com/komalsrathi> @dmiller15 <https://github.com/dmiller15>
> - I will merge this PR soon, because it is required to generate v8 JSONL
> files. Let me know if you have any review in progress, question,
> suggestion, or comment.
>
> cc @jharenza <https://github.com/jharenza>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/93#issuecomment-904785851>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZN26BDAMQI7QNGY7SRD6TT6PBBFANCNFSM5CQ3XF6A>
> .
>

komalsrathi commented:
@logstar if you are happy with the current review please go ahead as I am working on PNOC008 report generation right now.

logstar commented:
> @logstar, I don’t have any additional reviews.
> […](#)
> On Tue, Aug 24, 2021 at 12:19 PM Yuanchao Zhang ***@***.***> wrote: Hi @ewafula <https://github.com/ewafula> @komalsrathi <https://github.com/komalsrathi> @dmiller15 <https://github.com/dmiller15> - I will merge this PR soon, because it is required to generate v8 JSONL files. Let me know if you have any review in progress, question, suggestion, or comment. cc @jharenza <https://github.com/jharenza> — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#93 (comment)](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/93#issuecomment-904785851)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAZN26BDAMQI7QNGY7SRD6TT6PBBFANCNFSM5CQ3XF6A> .

@ewafula Thank you for the quick reply!

logstar commented:
> @logstar if you are happy with the current review please go ahead as I am working on PNOC008 report generation right now.

@komalsrathi Thank you for the quick reply!

I think this PR is ready to be used for v8 JSONL file generation, as it is has been tested on `snv-frequencies` and `cnv-frequencies`, which use `annotator` API and CLI respectively. If there is any error encountered during v8 JSONL file generation, I will submit another PR to fix.

However, it would still be very helpful to have your review on the R code changes.

komalsrathi commented:
@logstar reviewing now, get back to you soon!

komalsrathi reviewed (approved):
Hi @logstar I reran your module as well as tumor-gtex-plots and everything works perfectly, no duplicates, whitespaces or any other QC issues. Thank you!

logstar commented:
> Hi @logstar I reran your module as well as tumor-gtex-plots and everything works perfectly, no duplicates, whitespaces or any other QC issues. Thank you!

@komalsrathi Thank you for the review. I will merge this PR soon.
